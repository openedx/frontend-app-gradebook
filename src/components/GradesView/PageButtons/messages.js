import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  prevPage: {
    id: 'gradebook.GradesView.PageButtons.prevPage',
    defaultMessage: 'Previous Page',
    description: 'Grades tab Previous Page button text',
  },
  nextPage: {
    id: 'gradebook.GradesView.PageButtons.nextPage',
    defaultMessage: 'Next Page',
    description: 'Grades tab Next Page button text',
  },
});

export default messages;
