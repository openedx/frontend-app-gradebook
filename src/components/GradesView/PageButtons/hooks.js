import { useIntl } from '@edx/frontend-platform/i18n';

import { selectors, thunkActions } from 'data/redux/hooks';
import messages from './messages';

export const usePageButtonsData = () => {
  const { formatMessage } = useIntl();

  const { nextPage, prevPage } = selectors.grades.useGradeData();
  const getPrevNextGrades = thunkActions.grades.useFetchPrevNextGrades();

  const getPrevGrades = () => {
    getPrevNextGrades(prevPage);
  };

  const getNextGrades = () => {
    getPrevNextGrades(nextPage);
  };

  return {
    prev: {
      disabled: !prevPage,
      onClick: getPrevGrades,
      text: formatMessage(messages.prevPage),
    },
    next: {
      disabled: !nextPage,
      onClick: getNextGrades,
      text: formatMessage(messages.nextPage),
    },
  };
};

export default usePageButtonsData;
