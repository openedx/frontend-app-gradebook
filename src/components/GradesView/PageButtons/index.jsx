import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Button } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';
import messages from './messages';

export class PageButtons extends React.Component {
  constructor(props) {
    super(props);
    this.getPrevGrades = this.getPrevGrades.bind(this);
    this.getNextGrades = this.getNextGrades.bind(this);
  }

  getPrevGrades() {
    this.props.getPrevNextGrades(this.props.prevPage);
  }

  getNextGrades() {
    this.props.getPrevNextGrades(this.props.nextPage);
  }

  render() {
    return (
      <div
        className="d-flex justify-content-center"
        style={{ paddingBottom: '20px' }}
      >
        <Button
          style={{ margin: '20px' }}
          variant="outline-primary"
          disabled={!this.props.prevPage}
          onClick={this.getPrevGrades}
        >
          <FormattedMessage {...messages.prevPage} />
        </Button>
        <Button
          style={{ margin: '20px' }}
          variant="outline-primary"
          disabled={!this.props.nextPage}
          onClick={this.getNextGrades}
        >
          <FormattedMessage {...messages.nextPage} />
        </Button>
      </div>
    );
  }
}

PageButtons.defaultProps = {
  nextPage: '',
  prevPage: '',
};

PageButtons.propTypes = {
  // redux
  getPrevNextGrades: PropTypes.func.isRequired,
  nextPage: PropTypes.string,
  prevPage: PropTypes.string,
};

export const mapStateToProps = (state) => ({
  nextPage: selectors.grades.nextPage(state),
  prevPage: selectors.grades.prevPage(state),
});

export const mapDispatchToProps = {
  getPrevNextGrades: thunkActions.grades.fetchPrevNextGrades,
};

export default connect(mapStateToProps, mapDispatchToProps)(PageButtons);
