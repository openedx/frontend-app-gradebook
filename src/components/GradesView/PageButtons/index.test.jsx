import React from 'react';
import { shallow } from 'enzyme';

import { Button } from '@edx/paragon';

import usePageButtonsData from './hooks';
import PageButtons from '.';

jest.mock('./hooks', () => jest.fn());

const hookProps = {
  prev: {
    disabled: 'prev-disabled',
    onClick: jest.fn().mockName('hooks.prev.onClick'),
    text: 'prev-text',
  },
  next: {
    disabled: 'next-disabled',
    onClick: jest.fn().mockName('hooks.next.onClick'),
    text: 'next-text',
  },
};
usePageButtonsData.mockReturnValue(hookProps);

let el;
describe('PageButtons component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    el = shallow(<PageButtons />);
  });
  describe('behavior', () => {
    it('initializes component hooks', () => {
      expect(usePageButtonsData).toHaveBeenCalled();
    });
  });
  describe('render', () => {
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
    });
    test('prev button', () => {
      const button = el.find(Button).at(0);
      expect(button.props().disabled).toEqual(hookProps.prev.disabled);
      expect(button.props().onClick).toEqual(hookProps.prev.onClick);
      expect(button.text()).toEqual(hookProps.prev.text);
    });
    test('next button', () => {
      const button = el.find(Button).at(1);
      expect(button.props().disabled).toEqual(hookProps.next.disabled);
      expect(button.props().onClick).toEqual(hookProps.next.onClick);
      expect(button.text()).toEqual(hookProps.next.text);
    });
  });
});
