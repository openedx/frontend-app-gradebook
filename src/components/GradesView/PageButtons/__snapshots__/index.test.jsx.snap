// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageButtons component render snapshot 1`] = `
<div
  className="d-flex justify-content-center"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  <Button
    disabled="prev-disabled"
    onClick={[MockFunction hooks.prev.onClick]}
    style={
      Object {
        "margin": "20px",
      }
    }
    variant="outline-primary"
  >
    prev-text
  </Button>
  <Button
    disabled="next-disabled"
    onClick={[MockFunction hooks.next.onClick]}
    style={
      Object {
        "margin": "20px",
      }
    }
    variant="outline-primary"
  >
    next-text
  </Button>
</div>
`;
